### DISCUSSION TITLE
marimo.io file uploads

### DISCUSSION THREAD

**Discussion by pyinstallerbaggins** [marimo-team/marimo#2220] [Q&A]
I am using marimo.io for the access permision controls for admin and users 

How do you upload existing folders or files from your local drive? I click on upload file and it brings me to a directory but no files are populated to select. I

**Comment by akshayka** [comment]
There are two options, but I think it may be buggy, I'm also having difficulty.

1. Upload files from manage project:

![image](https://github.com/user-attachments/assets/b46be496-d478-4ede-938a-72857cf89172)

Is this what you tried? I just tried it, and some filetypes work (`.mp4` worked), but when I try to upload a CSV, nothing happens, even after refreshing. 

I'm not sure why no files were populated for you to select. Can you share screenshots? cc @mscolnick 

2. Upload files to the data catalog. These files can be accessed in any notebook:

![image](https://github.com/user-attachments/assets/90871ffe-0b5a-4e19-a5da-c1c60ca9ebcb)

**Reply by pyinstallerbaggins** [reply]
Trying but marimo.io/@public is down. Maybe down for maintenance? Ill try again later today 

edit: its back online
i see files in there now. fyi i didnt take any action - i believe it took its time to upload from the first try but it seem like nothing was being processed because is multiple letsgo.py files when my local drive only has one letsgo.py

Maybe add a spinning timer to notify its being loaded. 
![image](https://github.com/user-attachments/assets/fc92ae46-0851-415c-855e-fb0a129d0a81)

Secondly, Is there a way to upload whole folders or is it manually creating folders then uploading files ?

Lastly, i click open file on a marimo file letsgo.py and i expect it to populate on the right but nothing happens but it shows the content on the left

![image](https://github.com/user-attachments/assets/c67a60b1-f87d-47cd-849f-78f1efd59318)

After I click open file
![image](https://github.com/user-attachments/assets/ca02c5a7-6eb9-40bd-8eb2-288b0b8b22a1)



thanks for Marimo!
